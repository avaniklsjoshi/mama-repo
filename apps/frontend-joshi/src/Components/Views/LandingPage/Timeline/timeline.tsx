import { useEffect, useState } from "react";
import "./timeline.scss";
import { IconButton, Tooltip } from "@mui/material";
import { useTranslation } from "react-i18next";
import InfoButton from "./infoButton";
import { InfoIcon } from "../../../../Assets/SVG/infoIcon";
import { LANGUAGE_NAMES_KEY } from "../../../../Configs/Constants/const";
import {
  EXPERIENCE_TIMELINE_de,
  EXPERIENCE_TIMELINE_en,
  EXPERIENCE_TIMELINE_hn
} from "../../../../Configs/Constants/constWithTranslation";

export default function Timeline() {
  const { i18n } = useTranslation();
  const [activeTimeline, setActiveTimeline] = useState(EXPERIENCE_TIMELINE_en);

  useEffect(() => {
    if (i18n.language == LANGUAGE_NAMES_KEY.ENGLISH) {
      setActiveTimeline(() => EXPERIENCE_TIMELINE_en);
    } else if (i18n.language == LANGUAGE_NAMES_KEY.GERMAN) {
      setActiveTimeline(() => EXPERIENCE_TIMELINE_de);
    } else {
      setActiveTimeline(() => EXPERIENCE_TIMELINE_hn);
    }
  }, [i18n.language]);

  return (
    <div className="timeline">
      <div className="timeline-title">HOW IT ALL STARTED {i18n.language}</div>

      {activeTimeline &&
        activeTimeline.map((exp: any) => (
          <div
            className="timeline__event  animated fadeInUp delay-3s timeline__event--type1"
            key={exp.title}
          >
            <div className="timeline__event__icon ">
              <i className="lni-cake">
                <img src={exp.thumbnail} alt="jiji" />
              </i>
            </div>
            <div className="timeline__event__date">{exp.span}</div>
            <div className="timeline__event__content ">
              <div className="timeline__event__title">
                {exp.title}
                <Tooltip
                  title={
                    <InfoButton content={exp.info} projects={exp.projects} />
                  }
                  className="info-icon"
                >
                  <IconButton>
                    <InfoIcon className="feature-iicon" />
                  </IconButton>
                </Tooltip>
              </div>
              <div className="timeline__event__description">
                <p>Projects- {exp.projects}</p>
                <p>Role- {exp.role}</p>
              </div>
            </div>
          </div>
        ))}
    </div>
  );
}
