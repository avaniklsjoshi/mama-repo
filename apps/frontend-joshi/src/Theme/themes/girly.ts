export default {
  themeType: "girly",
  colors: {
    background: "#f1b7ea",
    text: "#fff"
  },
  header: {
    background: "#000",
    text: "#fff"
  },
  footer: {
    background: "#262726",
    text: "#fff"
  },
  section: {
    background: "#008F7A",
    text: "#fff"
  }
};
