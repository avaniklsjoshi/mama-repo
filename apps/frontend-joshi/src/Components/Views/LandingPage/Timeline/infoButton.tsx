import "./timeline.scss";

interface InfoButtonProp {
  projects: string;
  content: { tech: string[]; role: string[] };
}

export default function InfoButton(props: InfoButtonProp) {
  const isProjects = props.projects.split(",").length > 1;
  return (
    <div className="info-container">
      <div className="tech-container">
        <div className="info-title">
          Tech Stacks used in {isProjects ? "these projects" : "this project"} -
        </div>
        <ul>
          {props.content.tech.map((tech: string, index: number) => (
            <li key={index}>{tech}</li>
          ))}
        </ul>
      </div>
      <div className="role-container">
        <div className="info-title">
          My Role in {isProjects ? "these projects" : "this project"}-
        </div>
        <ul>
          {props.content.role.map((role: string, index: number) => (
            <li key={index}>{role}</li>
          ))}
        </ul>
      </div>
    </div>
  );
}
