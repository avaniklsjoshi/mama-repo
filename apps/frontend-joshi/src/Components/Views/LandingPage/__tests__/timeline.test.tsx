/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import girly from "../../../../Theme/themes/girly";
import Timeline from "../Timeline/timeline";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: any) => key,
    i18n: (language: any) => language
  })
}));

describe("Landing Page", () => {
  it("renders Timeline", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <Timeline />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("timelineTitle")).toBeInTheDocument();
  });
});
