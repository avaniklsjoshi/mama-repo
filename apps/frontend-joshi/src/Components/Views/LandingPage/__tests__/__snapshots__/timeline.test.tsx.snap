// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page renders Timeline 1`] = `
<div>
  <div
    class="timeline"
  >
    <div
      class="sc-bczRLJ jliZnU timeline-title"
    >
      timelineTitle
    </div>
    <div
      class="timeline__event  animated fadeInUp delay-3s timeline__event--type1"
    >
      <div
        class="timeline__event__icon "
      >
        <i
          class="lni-cake"
        >
          <img
            alt="jiji"
            src="../../../assets/Images/sauropod-svgrepo-com.svg"
          />
        </i>
      </div>
      <div
        class="timeline__event__date"
      >
        12-09-2019 - अब तक
      </div>
      <div
        class="timeline__event__content "
      >
        <div
          class="timeline__event__title"
        >
          एटेक्चर जीएमबीएच, सॉफ्टवेयर डेवलपर
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium info-icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              height="14px"
              id="Capa_1"
              version="1.1"
              viewBox="0 0 111.577 111.577"
              width="14px"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <g>
                <path
                  d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                  fill="#2a272763"
                />
              </g>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="timeline__event__description"
        >
          <p>
            projects
            - 
            VW AutoSuche, VW Onehub, ING db GeldAutomat
          </p>
          <p>
            role
            - 
            डिजाइनिंग आर्किटेक्चर, ऐप्स की बेसिक सेटिंग, इम्प्लीमेंटेशन
          </p>
        </div>
      </div>
    </div>
    <div
      class="timeline__event  animated fadeInUp delay-3s timeline__event--type1"
    >
      <div
        class="timeline__event__icon "
      >
        <i
          class="lni-cake"
        >
          <img
            alt="jiji"
            src="../../../Assets/Images/flower-svgrepo-com.svg"
          />
        </i>
      </div>
      <div
        class="timeline__event__date"
      >
        09/2017 – 05/2019
      </div>
      <div
        class="timeline__event__content "
      >
        <div
          class="timeline__event__title"
        >
          टिकटमास्टर, फ्रंटएंड डेवलपर
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium info-icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              height="14px"
              id="Capa_1"
              version="1.1"
              viewBox="0 0 111.577 111.577"
              width="14px"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <g>
                <path
                  d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                  fill="#2a272763"
                />
              </g>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="timeline__event__description"
        >
          <p>
            projects
            - 
            CMS, Virtual Venue
          </p>
          <p>
            role
            - 
            डिवाइस और ब्राउज़र संगत ऐप्स बनाना
          </p>
        </div>
      </div>
    </div>
    <div
      class="timeline__event  animated fadeInUp delay-3s timeline__event--type1"
    >
      <div
        class="timeline__event__icon "
      >
        <i
          class="lni-cake"
        >
          <img
            alt="jiji"
            src="../../Assets/Images/dinosaur-svgrepo-com.svg"
          />
        </i>
      </div>
      <div
        class="timeline__event__date"
      >
        02/15 – 07/17
      </div>
      <div
        class="timeline__event__content "
      >
        <div
          class="timeline__event__title"
        >
          वेबसेनोर टेक्नोलॉजीज, फुलस्टैक डेवलपर
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium info-icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              height="14px"
              id="Capa_1"
              version="1.1"
              viewBox="0 0 111.577 111.577"
              width="14px"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <g>
                <path
                  d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                  fill="#2a272763"
                />
              </g>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="timeline__event__description"
        >
          <p>
            projects
            - 
            नोवा फ़ार्मा
          </p>
          <p>
            role
            - 
            आवश्यकता एकत्र करना, डिजाइन करना और कार्यान्वयन करना.
          </p>
        </div>
      </div>
    </div>
    <div
      class="timeline__event  animated fadeInUp delay-3s timeline__event--type1"
    >
      <div
        class="timeline__event__icon "
      >
        <i
          class="lni-cake"
        >
          <img
            alt="jiji"
            src="../../Assets/Images/flower-green.svg"
          />
        </i>
      </div>
      <div
        class="timeline__event__date"
      >
        09/14 – 12/14
      </div>
      <div
        class="timeline__event__content "
      >
        <div
          class="timeline__event__title"
        >
          इंफोसिस लिमिटेड, प्रशिक्षु
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium info-icon css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              height="14px"
              id="Capa_1"
              version="1.1"
              viewBox="0 0 111.577 111.577"
              width="14px"
              x="0px"
              xml:space="preserve"
              y="0px"
            >
              <g>
                <path
                  d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                  fill="#2a272763"
                />
              </g>
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="timeline__event__description"
        >
          <p>
            projects
            - 
            शॉपिंग पोर्टल
          </p>
          <p>
            role
            - 
            Microsoft .NET तकनीक सीखना
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
